core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1457,6 +1465,18 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
     }
   }
 
+  /**
+   * Validate some configurations for new MetadataVersion. A new MetadataVersion can take place when
+   * a FeatureLevelRecord for "metadata.version" is read from the cluster metadata.
+   */
+  def validateWithMetadataVersion(metadataVersion: MetadataVersion): Unit = {
```

Comment:
This seems to be the exact same check we do in `validateValues()`. Can we consolidate both?

Reply:
We can call this from `validateValues`, but then I won't be able to take your other suggestion to mention `inter.broker.protocol.version` in the error message. This one gets called when the KRaft broker discovers a new metadata.version via a new feature record. Would you prefer we consolidate the checks, or keep mention of `inter.broker.protocol.version` when it makes sense? 
I lean towards leaving them both, so it's helpful for the operator, and the duplication cost should go away in 4.0.

Reply:
Ah right, thanks for the explanation. Then I guess it makes sense to keep both methods for now.